Improve localhost SSL integration docs [ci skip] (#2712)
puma/puma: README.md

```
$ puma -b 'ssl://127.0.0.1:9292?key=path_to_key&cert=path_to_cert'
```
#### Self-signed SSL certificates (via the [`localhost`] gem, for development use): 

Puma supports the [`localhost`] gem for self-signed certificates. This is particularly useful if you want to use Puma with SSL locally, and self-signed certificates will work for your use-case. Currently, the integration can only be used in MRI. 

Puma will automatically configure SSL if you require the [`localhost`] gem when running in environment `development`:

```ruby
# Add the localhost gem to your Gemfile
group(:development) do 
  gem 'localhost'
end

> # config.ru:

Reviewer: I was pretty confused by this, maybe it would make more sense as a sentence?

```suggestion
# Alternatively, you can require the gem in config.ru:
```

# Require it implicitly using bundler
require "bundler"
Bundler.require(:default, ENV["RACK_ENV"].to_sym)

# Or require it explicitly
require './app'
require 'localhost'
run Sinatra::Application
```

> You also need to make sure Puma listens to an SSL socket:

Reviewer: I think we can be a bit more concise here:

```suggestion
Additionally, Puma must be listening to an SSL socket:
```

```shell
$ puma -b 'ssl://localhost:9292' config.ru

> # You can still have Puma being reachable over HTTP by repeating
> # the -b flag (tcp://), but you need to use a different port:
> $ puma -b ssl://localhost:9292 -b tcp://localhost:9393 config.ru

Reviewer: I think this makes a little bit more sense. What do you think?

```suggestion
# The following options allow you to reach Puma over HTTP as well:
$ puma -b ssl://localhost:9292 -b tcp://localhost:9393 config.ru
```
